### DISCUSSION TITLE
[4.x] Add Lule Sami language code for sites

### DISCUSSION THREAD

**Discussion by pappgergely** [craftcms/cms#12703] [Ideas] [answered]
A client requested adding the Lule Sami language (https://en.wikipedia.org/wiki/Lule_S%C3%A1mi) to a section of their site.

When adding a new site to Craft CMS, I can only select from predefined values, and I cannot add a new language code myself.

Please add Lule Sami as a selectable value for the site language.

**Comment by brandonkelly** [comment, marked as answer]
Craft’s locale data comes from the [ICU library](https://icu.unicode.org/) (pulled in via PHP’s [Intl](https://www.php.net/manual/en/intro.intl.php) extension), which is the industry standard for application localization.

Based on [this chart](https://cldr.unicode.org/index/locale-coverage) I see that Lule Sámi is not included in ICU yet. According to that Wikipedia article the language only has 650 speakers and declining, so it’s unlikely it ever will be.

So I’ve just submitted a PR that makes it possible to add custom locale aliases to Craft (#12705). That would at least allow you to select Lule Sámi as a site language. Unfortunately this is a semi-breaking change so has to wait until Craft 5.0 to be released, though.
